backend/app/api/quiz.py

```
@@ -0,0 +1,71 @@
+    """
+    Get a single quiz by its ID.
+    """
+    db_quiz = await crud_quiz.get_quiz(session=session, quiz_id=quiz_id)
+    if not db_quiz:
+        raise HTTPException(status_code=404, detail="Quiz not found")
+    return db_quiz
+
+@router.patch("/{quiz_id}", response_model=quiz_schemas.QuizRead)
+async def update_quiz(
+    *,
+    session: Session = Depends(get_db),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)